### ISSUE TITLE
`TypeError: Cannot read property 'concat' of undefined` when running ESLint via Jest

### ISSUE THREAD

**Issue by elliottsj** [eslint/eslint#5791]
This error occurs when running ESLint from a Jest test:

```sh
TypeError: Cannot read property 'concat' of undefined
    at Object.<anonymous> (/Users/spencerelliott/Sites/elliottsj/eslint-jest-issue/node_modules/eslint/lib/util/module-resolver.js:28:30)
    at Object.<anonymous> (/Users/spencerelliott/Sites/elliottsj/eslint-jest-issue/node_modules/eslint/lib/config/config-file.js:23:22)
    at Object.<anonymous> (/Users/spencerelliott/Sites/elliottsj/eslint-jest-issue/node_modules/eslint/lib/config.js:17:18)
    at Object.<anonymous> (/Users/spencerelliott/Sites/elliottsj/eslint-jest-issue/node_modules/eslint/lib/cli-engine.js:31:14)
    at Object.<anonymous> (/Users/spencerelliott/Sites/elliottsj/eslint-jest-issue/node_modules/eslint/lib/api.js:10:16)
    at Object.<anonymous> (/Users/spencerelliott/Sites/elliottsj/eslint-jest-issue/src/index.js:1:136)
    at Object.<anonymous> (/Users/spencerelliott/Sites/elliottsj/eslint-jest-issue/src/__tests__/index-test.js:3:12)
    at jasmine2 (/Users/spencerelliott/Sites/elliottsj/eslint-jest-issue/node_modules/jest-jasmine2/src/index.js:154:16)
```

**What version of ESLint are you using?**
[`eslint@2.7.0`](https://github.com/elliottsj/eslint-jest-issue/blob/e7a0a3c60ed5dff81cacd29a10a6184a832ca732/package.json#L20)

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
https://github.com/elliottsj/eslint-jest-issue/blob/master/.eslintrc

```
{
  "extends": "eslint:recommended",
  "env": {
    "jest": true,
    "node": true
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

See the full example here: https://github.com/elliottsj/eslint-jest-issue

```js
// index.js
var CLIEngine = require('eslint').CLIEngine;

function lint() {
  var cli = new CLIEngine();

  // Lint this file
  var report = cli.executeOnFiles(['index.js']);

  return report;
}

module.exports = {
  lint: lint
};
```

```js
// __tests__/index-test.js
jest.unmock('..');

var lint = require('..').lint;

describe('eslint CLIEngine', function() {
  it('should work with Jest\'s module system', function() {
    var report = lint();
    expect(report.results).toBeDefined();
  });
});
```

**What did you expect to happen?**

Expected the Jest test to pass without error.

**What actually happened? Please include the actual, raw output from ESLint.**

See error shown above.

---

ESLint v2.5.3 runs fine: https://github.com/elliottsj/eslint-jest-issue/tree/eslint-2.5

It seems the issue was introduced as a result of https://github.com/eslint/eslint/commit/4ecd45e549c8622a26f5b105dd43b7a2827cf942: here is the failing line: https://github.com/eslint/eslint/commit/4ecd45e549c8622a26f5b105dd43b7a2827cf942#diff-6535c964600bdcc5abbfefe8282e96f3R28

**Comment by nzakas** [comment]
@elliottsj can you please fill in questions you left empty above? It really helps us to better triage (many of us check in on phones and tablets, so having all the info in the issue lets us be a lot more efficient than needing to go to a different repo).

**Comment by elliottsj** [comment]
@nzakas Sorry if it was unclear; for the first four questions, you can take a look at the example project: https://github.com/elliottsj/eslint-jest-issue

I'll fill in the answers for clarity

**Comment by nzakas** [comment]
@elliottsj thanks. It's a lot easier when the questions are answered in the issue.

What version of Node are you using?

**Comment by elliottsj** [comment]
@nzakas Node v4.4.1

**Comment by nzakas** [comment]
Hmm, that's strange. It looks like `module.paths` is missing for some reason, and as far as I know, that's not possible. Ill have to investigate a bit

**Comment by nzakas** [comment]
Looking at this now.

**Comment by nzakas** [comment]
So this looks like an issue with the way Jest is loading everything. I can put in a check that will avoid the error you're getting, but I'm not sure if you'll get other errors as a result. I'd suggest filing a bug with Jest to continue this discussion.

**Comment by elliottsj** [comment]
I've opened an issue with Jest: https://github.com/facebook/jest/issues/894

**Comment by cpojer** [comment]
This was fixed in Jest 11.0.

**Comment by cpojer** [comment]
@nzakas Is there a reason why eslint is using module.paths at all? It seems like it is undocumented private API that is an implementation detail of the node module resolution algorithm.

Jest sandboxes all JavaScript code for proper test isolation (that way no module or test can impact another test). We are using node-resolve for all resolution which has a userland implementation of what `module.paths` gives you. I suggest to use it instead: https://github.com/substack/node-resolve
